logsapi/server.go

```diff
@@ -64,14 +64,22 @@ func handler(libhoneyClient *libhoney.Client) http.HandlerFunc {
 			switch v := msg.Record.(type) {
 			case string:
 				// attempt to parse as json
-				var record map[string]interface{}
-				err := json.Unmarshal([]byte(v), &record)
+				var jsonRecord map[string]interface{}
+				err := json.Unmarshal([]byte(v), &jsonRecord)
 				if err != nil {
+					// not JSON; we'll treat this log entry as a timestamped string
 					event.Timestamp = parseMessageTimestamp(event, msg)
 					event.AddField("record", msg.Record)
 				} else {
-					event.Timestamp = parseFunctionTimestamp(msg, record)
-					event.Add(record)
+					// we've got JSON
+					event.Timestamp = parseFunctionTimestamp(msg, jsonRecord)
+					if jsonRecord["data"] != nil {
```

Comment:
is it worth checking to see if this is a map? Something like:

    switch v := jsonRecord["data"].(type) {
    case map[string]interface{}:
             // data key contains a map, so add fields from it
             event.Add(jsonRecord["data"])
    default:
             // data is not a map, so just add the record as if it were a flat JSON object
             event.Add(jsonRecord)
    }

Reply:
Yes! That would be more better.